File: workout-tracker-app-webservice/backend/src/main/java/hu/bme/aut/workout_tracker_backend/data_layer/exercise/Exercise.java
package hu.bme.aut.workout_tracker_backend.data_layer.exercise;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Exercise {
    @Id
    @GeneratedValue(strategy= GenerationType.AUTO, generator="native")
    @GenericGenerator(name = "native")
    private Long id;
    private String category;
    private String name;
}

File: workout-tracker-app-webservice/app/app/src/main/java/hu/bme/aut/workout_tracker/data/model2/Exercise.kt
package hu.bme.aut.workout_tracker.data.model2


import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Exercise(
    @Json(name = "id")
    val id: Int,
    @Json(name = "category")
    val category: String,
    @Json(name = "name")
    val name: String
)